### ISSUE TITLE
Toggle breakpoints by single click

### ISSUE THREAD

**Issue by timhoffm** [spyder-ide/spyder#5005]
The behavior for setting breakpoints by mouse is somewhat unusual compared to other IDEs.

**What is the expected output? What do you see instead?**

**Current behavior:**

- Single-click on line number: Select row
- Single-click on icon area (left of line number): Toggle: Select row
- Double-click on line number or icon area: Toggle breakpoint

**Desired behavior:**

- Single-click on line number: Select row
- Single-click on icon area: Toggle: **Toggle breakpoint**
- Double-click on line number or icon area: **(none)**

The desired behavior is the behavior of every IDE I've worked with so far (QtCreator, Eclipse, PyCharm, ...)

If you agree to change this, I'll provide a pull request.

## Version and main components

* Spyder Version: 3.2.2.dev0 
* Python Version: 3.6.2
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.3 (OK)
numpy >=1.7      :  1.13.1 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.26 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.1.0 (OK)
pylint >=0.25    :  1.7.2 (OK)

```

**Comment by ccordoba12** [comment]
@rlaverde, what do you think?

**Comment by goanpeca** [comment]
It is the default behavior on some ides so :+1: although I dont mind either way tbh :-p

**Comment by ccordoba12** [comment]
My opinion is that we should leave this for Spyder 4 with the new *debugging* panel that @rlaverde is going to add to the Editor. That's why I asked him.

**Comment by rlaverde** [comment]
I agree with @timhoffm, double click is somehow unnecessary, I'll take this in account when implementing #528

**Comment by ccordoba12** [comment]
Thanks @rlaverde!

**Comment by timhoffm** [comment]
Thanks you for your replies. I'm not quite happy to hear that you want to defer this to Spyder 4. 😞  Can you please please motivate, why you want to defer this until Spyder 4. Also, is there an estimate, when Spyder 4 might come out?

**Comment by ccordoba12** [comment]
> Can you please please motivate, why you want to defer this until Spyder 4

Because we don't want to keep adding/modifying features in our maintenance branch, else we won't be able to stabilize it.

> Also, is there an estimate, when Spyder 4 might come out?

June 2018.

**Comment by jnsebgosselin** [comment]
@ccordoba12 I've checked and single-click to add a break point is also the default behaviour in Matlab.

@timhoffm While waiting for Spyder4 to implement this feature, you can patch your local version of Spyder3 to get the behaviour you wish by modifying the [linenumberarea_mousepress_event ](https://github.com/spyder-ide/spyder/blob/3.x/spyder/widgets/sourcecode/codeeditor.py#L1263) method in the [codeeditor](https://github.com/spyder-ide/spyder/blob/3.x/spyder/widgets/sourcecode/codeeditor.py) as follow:

```
def linenumberarea_mousepress_event(self, event):
    """Handling line number area mouse double press event"""
    line_number = self.__get_linenumber_from_mouse_event(event)
    self.linenumberarea_pressed = line_number
    self.linenumberarea_released = line_number
    if event.x() <= self.get_markers_margin():
        self.linenumberarea_mousedoubleclick_event(event)
    else:
        self.linenumberarea_select_lines(self.linenumberarea_pressed, 
                                         self.linenumberarea_released)
```
**Resulting behavior:**

Single-click on line number: Select row
Single-click on icon area: Toggle: Toggle breakpoint
Double-click on line number or icon area:  Toggle breakpoint

**Comment by goanpeca** [comment]
@ccordoba12 this seems like a small enough "feature" to be included in a patch release. I would not mind that.

**Comment by ccordoba12** [comment]
Sorry but this implies not only a change in code but also in user experience. And this something *nobody* has complained about in all my years in Spyder (until now :-).

So this is for Spyder 4.

**Comment by goanpeca** [comment]
>And this something nobody has complained about in all my years in Spyder (until now :-).

Everyone has complained about the debugger... so it is kinda implied

**Comment by ccordoba12** [comment]
Probably, but nobody has mentioned this particular problem explicitly. However, I've heard multiple complaints about our other problems.

**Comment by goanpeca** [comment]
Sure ;-) just my 2 cents

**Comment by timhoffm** [comment]
@jnsebgosselin I like your change. It's the minimal change in user experience for the maximal benefit in user expectations. 😄 (I assume that only few people will currently use single-clicking on the icon area for row selection in Spyder. But people coming from other IDEs will likely expect setting breakpoints by single-click).

I would be happy to see this in 3.x, but just another 2 cents.
